Formula/dpkg.rb

```diff
@@ -16,13 +16,19 @@ class Dpkg < Formula
 
   depends_on "pkg-config" => :build
   depends_on "gnu-tar"
+  depends_on "gpatch"
   depends_on "xz" # For LZMA
 
   def install
     # We need to specify a recent gnutar, otherwise various dpkg C programs will
     # use the system "tar", which will fail because it lacks certain switches.
     ENV["TAR"] = Formula["gnu-tar"].opt_bin/"gtar"
 
+    # Since 1.18.24 dpkg mandates the use of GNU patch to prevent occurrences
+    # of the CVE-2017-8283 vulnerability.
+    # http://www.openwall.com/lists/oss-security/2017/04/20/2
+    ENV["PATCH"] = Formula["gpatch"].opt_bin/"patch"
```

Comment:
Hey, upstream here. :)

I was  not very happy about relying on GNU patch here, but given that dpkg release cycle is tied to the Debian release cycle, and the latter is in deep freeze right now, implementing the required checks was deemed too invasive. I'm planning on doing that during the dpkg 1.19.x cycle, which I just opened up several days ago on git master.

Re the environment variable, that's the common autotools way to override the default tools. If you find that unpleasant, or suboptimal for some reason, and rather use a configure option or similar I can accommodate. ;)

Re GNU tar, I'm also unhappy about that being a requirement, and my plan is to replace that with an builtin implementation, in principle during the 1.19.x cycle.

Re portability in general, dpkg has always been somewhat portable, but during my maintainership I've tried to improve that substantially, as I consider portability a very important aspect (you can see from git log or debian/changlog :). One problem I had before was that I didn't have access to a Mac OS X / macOS system, now that I do thanks to the Fink folks, I've fixed several issues to better support the system. Obviously there is still some work to do, for example to support Mach-O in the perl code, etc. But for example start-stop-daemon and update-alternatives now work and the latter passes the test suite.

I also tend to check downstream ports/pkgs to see what are the current pain points, or things that could be fixed upstream etc. See https://wiki.debian.org/Teams/Dpkg/Downstream and https://wiki.debian.org/Teams/Dpkg/Porting. In any case in the future I'd encourage you to let me know of this kind of things (either on the mailing list or on IRC), I'm very receptive to portability and downstream issues!